src/components/request/RequestForm.tsx

```diff
@@ -194,20 +215,76 @@ const RequestForm: FC = () => {
     }
   }, []);
 
+  // Temp. This throws if you try to read certain values too quickly.
+  // TODO: Add proper loader.
+  useEffect(() => {
+    setTimeout(() => {
+      setRaceCond(true);
+    }, 2000);
+  }, []);
+
   return (
     <RequestFormWrapper>
       <RequestFormRow>
         <RequestFormHeaderAndFormWrapper>
-          <FormHeader>Proposal</FormHeader>
+          <FormHeader>
+            {flags.InProposeState && "Proposal"}
+            {(flags.InDisputeState || flags.DisputeInProgress) &&
+              "Dispute Period"}
+          </FormHeader>
           <RequestFormInputWrapper>
             <RequestInputButtonBlock>
-              <RequestFormInput
-                label="Propose: "
-                value={value}
-                onChange={inputOnChange}
-              />
+              {flags.InProposeState && (
+                <RequestFormInput
+                  label="Propose: "
+                  value={value}
+                  onChange={inputOnChange}
+                />
+              )}
+              {flags.InDisputeState && raceCond && (
+                <RequestFormInput
+                  disabled={true}
+                  label="Proposed answer: "
+                  value={ethers.utils.formatUnits(
+                    read().request().proposedPrice,
+                    read().collateralProps().decimals
+                  )}
+                  onChange={() => null}
+                />
+              )}
               {getButton(value)}
             </RequestInputButtonBlock>
+            {inputError && <InputError>{inputError}</InputError>}
+            {flags.InDisputeState && (
+              <ProposerAddress>
+                Proposer:{" "}
+                <a
+                  target="_blank"
+                  rel="noreferrer"
+                  href={`${
+                    CHAINS[(state.inputs?.request?.chainId || 1) as ChainId]
```

Comment:
you could probably also export `explorerUrl`, or better yet the entire link string with addresses in the oracle reader hook directly. I would think of this like redux's [useSelector](https://react-redux.js.org/api/hooks#useselector), and to use it to format the raw state ( or in this case state reader) to something consumable in the view directly.  i think a rule of thumb might be, if you are doing logic in the view, move it to the hook for manipulation and import it into the view to use directly. let me know what u think

Reply:
Moved to useReader.